src/main.d

```diff
@@ -231,28 +293,6 @@ int main(string[] args)
 	log.vlog("Opening the item database ...");
 	auto itemdb = new ItemDatabase(cfg.databaseFilePath);
 	
-	// Set the local path OneDrive root
-	string syncDir;
-	if ((environment.get("SHELL") == "") && (environment.get("USER") == "")){
-		// No shell or user set, so expandTilde() will fail - usually headless system running under init.d / systemd
-		// Did the user specify a 'different' sync dir by passing a value in?
-		if (syncDirName){
-			// was there a ~ in the passed in state? it will not work via init.d / systemd
-			if (canFind(cfg.getValue("sync_dir"),"~")) {
-				// A ~ was found
-				syncDir = homePath ~ "/OneDrive";
-			} else {
-				// No ~ found in passed in state, use as is
-				syncDir = cfg.getValue("sync_dir");
-			}
-		} else {
-			// need to create a default as expanding ~ will not work
-			syncDir = homePath ~ "/OneDrive";
-		}
-	} else {
-		// A shell and user is set, expand any ~ as this will be expanded if present
-		syncDir = expandTilde(cfg.getValue("sync_dir"));
-	}
```

Comment:
I am thinking here that `--display-config` really needs to display the entire 'interpreted' configuration it will use, so if a user is using `~` in the configuration, the display will get extrapolated out. Moving the env / path section to below the configuration initialisation made sense as keeping it 'below' sync engine init etc did not make sense to me, and did not appear to have any 'flow' on impact with sync engine init